pkg/server/singleprocess/auth.go

```diff
@@ -459,8 +473,15 @@ func (s *Service) GenerateLoginToken(
 		}
 	}
 
+	encodedId, err := s.encodeId(ctx, currentUser.Id)
+	if err != nil {
+		msg := "failed to encode id when generating a login token"
+		log.Error(msg, "currentUser.Id", currentUser.Id, "err", err)
+		return nil, status.Error(codes.Internal, msg)
+	}
+
 	login := &pb.Token_Login{
-		UserId: currentUser.Id,
+		UserId: encodedId,
```

Comment:
I'll note that I don't love this flow. It's not at all intuitive when IDs need to be encoded/decoded, and I don't have any great heuristics to offer. It will be easy for us to implement new logic in the future that skips doing this.